### PR TITLE
[#3] Add custom font extention

### LINKED ISSUE #3: [UI] As a user, I can see a Custom Font
## Why

To allow the app to show a custom font.

## Acceptance Criteria

After #2, put a UILabel in the app to show the custom font with the text "Hello" size: 14 weight: regular.

Font: https://fonts.google.com/specimen/Zen+Old+Mincho#standard-styles

### PR DESCRIPTION
**Pull request by blyscuit**
close #3

## What happened

Add custom helper class.
 
## Insight

Use custom helper class for custom font size in the placeholder screen. 
 
## Proof Of Work

https://user-images.githubusercontent.com/6356137/140684504-7592e96e-dcdb-4bf1-ac4e-c4f76dd54fe5.mp4

### CODE REVIEW

**Comment by markgravity** [comment]
> 3 Warnings
> ⚠️	[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19) - **[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19)**: Will never be executed
> ⚠️	[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26) - **[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26)**: Will never be executed
> ⚠️	[CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41) - **[CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41)**: Will never be executed
> ## Current coverage for CustomDynamicFont is `39.10%`
> Files changed	-	-
> RequestConfiguration.swift	`0.00%`	💀
> Navigator.swift	`0.00%`	💀
> Color+Application.swift	`0.00%`	💀
> NetworkAPI.swift	`0.00%`	💀
> NetworkAPIProtocol.swift	`0.00%`	💀
> UIView+Subviews.swift	`0.00%`	💀
> UIFont+CustomFont.swift	`85.71%`	✅
> HomeViewController.swift	`100.00%`	✅
> Optional+Unwrap.swift	`100.00%`	✅
> AppDelegate.swift	`100.00%`	✅
> ZenOldMincho.swift	`100.00%`	✅
> > Powered by [xcov](https://github.com/nakiostudio/xcov)
> 
> Generated by 🚫 [Danger](https://danger.systems/)

Can we manage to clear these warnings please? 🙏

File: CustomDynamicFont/Sources/Presentation/CustomFont/ZenOldMincho.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            case .title2: return 22.0
+            case .title3: return 20.0
+            case .headline: return 17.0
+            case .body: return 17.0
+            case .callout: return 16.0
+            case .subheadline: return 15.0
+            case .footnote: return 13.0
+            case .caption1: return 12.0
+            case .caption2: return 1.0
+            default: return 17.0
```
Comment (minhnimble) [inline CustomDynamicFont/Sources/Presentation/CustomFont/ZenOldMincho.swift:39]: Optional: should we define enum representations for these float values like DeeMoney project as we maybe able to reuse them when there is new custom font? What do you think? 🤔

**Review by minhnimble — APPROVED** [review]
The rest LGTM 🚀

File: CustomDynamicFont/Sources/Presentation/CustomFont/ZenOldMincho.swift (hunk 1 above)
Comment (blyscuit) [reply CustomDynamicFont/Sources/Presentation/CustomFont/ZenOldMincho.swift:39]: Fonts do not have a same size when having the same points of font size. I think it doesn't make sense to include such enum. Also font scale varies from projects and font. I think we should treat this like project's color and not include a default value.
If default value is needed then we can use systemFont as a base.

**Comment by blyscuit** [comment]
> > 3 Warnings
> > ⚠️	[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19) - **[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L19)**: Will never be executed
> > ⚠️	[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26) - **[CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFont/Sources/Presentation/Navigator/Navigator.swift#L26)**: Will never be executed
> > ⚠️	[CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41) - **[CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41](https://github.com/nimblehq/ios-custom-font-scaling-demo/blob/ca25c8553e81fd49278220a8aeabed5b829bfda9/CustomDynamicFontTests/Sources/Mocks/Sourcery/AutoMockable.generated.swift#L41)**: Will never be executed
> > ## Current coverage for CustomDynamicFont is `39.10%`
> > Files changed	-	-
> > RequestConfiguration.swift	`0.00%`	💀
> > Navigator.swift	`0.00%`	💀
> > Color+Application.swift	`0.00%`	💀
> > NetworkAPI.swift	`0.00%`	💀
> > NetworkAPIProtocol.swift	`0.00%`	💀
> > UIView+Subviews.swift	`0.00%`	💀
> > UIFont+CustomFont.swift	`85.71%`	✅
> > HomeViewController.swift	`100.00%`	✅
> > Optional+Unwrap.swift	`100.00%`	✅
> > AppDelegate.swift	`100.00%`	✅
> > ZenOldMincho.swift	`100.00%`	✅
> > > Powered by [xcov](https://github.com/nakiostudio/xcov)
> > 
> > 
> > Generated by 🚫 [Danger](https://danger.systems/)
> 
> Can we manage to clear these warnings please? 🙏

These will be fix in #6 when the VC is implemented.

**Review by markgravity — APPROVED** [review]
LGTM

**Pull request merged**